pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -102,6 +103,37 @@ func Convert_v1_Disk_To_api_Disk(diskDevice *v1.Disk, disk *Disk, devicePerBus m
 	return nil
 }
 
+func setDriverCacheMode(disk *Disk, mode v1.DriverCache) error {
+	supportDirectIO := true
+
+	if mode == "" || mode == v1.CacheNone {
+		f, err := os.OpenFile(disk.Source.File, syscall.O_RDONLY|syscall.O_DIRECT, 0)
```

Comment:
this won't always work. We aren't guaranteed the disk is just a file anymore. It can also be a block device.

Reply:
yes, but I think it should work. I'm not entirely sure about the Go `os.OpenFile` itself, but opening a block device with O_DIRECT should be supported.

Reply:
with block device ```os.OpenFile``` will work, I will confirm that in functional tests.

Reply:
@davidvossel but you're right, I have to check ```Disk.Source.Dev``` as well